### PR TITLE
core: use atomic type

### PR DESCRIPTION
**Pull request by s7v7nislands**

### CODE REVIEW

**Review by holiman — APPROVED** [review]
LGTM!

File: core/blockchain.go [hunk 1]
```diff
@@ -215,8 +215,8 @@ type BlockChain struct {
 
 	wg            sync.WaitGroup //
 	quit          chan struct{}  // shutdown signal, closed in Stop.
-	running       int32          // 0 if chain is running, 1 when stopped
-	procInterrupt int32          // interrupt signaler for block processing
+	stopping      atomic.Bool    // false if chain is running, true when stopped
```
Comment (rjl493456442) [inline core/blockchain.go:218]: perhaps `stopped`?

**Review by rjl493456442 — APPROVED** [review]
Lgtm, one nitpick though.

File: core/blockchain.go (hunk 1 above)
Comment (holiman) [reply core/blockchain.go:218]: I think `stopping` is more correct -- it becomes `true` when `stopWithoutSaving` is invoked -- flipping this bool is basically the very first thing that happens on a call to `Stop()`. After this, all the waits happen. 

So this boolean does not mean it is `stopped`, but rather that the process of `stopping` the blockchain has (at least) begun. 

So IMO it's good the way it is

Comment (s7v7nislands) [reply core/blockchain.go:218]: Does the comment need to change to `true when stopping`?

**Pull request merged**